### ISSUE TITLE
allow customization of verification email links

### ISSUE THREAD

**Issue by josefaidt** [aws-amplify/amplify-backend#1150]
### Environment information

```plain text
System:
  OS: macOS 14.3.1
  CPU: (10) arm64 Apple M1 Pro
  Memory: 2.16 GB / 32.00 GB
  Shell: /opt/homebrew/bin/fish
Binaries:
  Node: 20.11.1 - ~/Library/Caches/fnm_multishells/76639_1710429675053/bin/node
  Yarn: 1.22.19 - ~/Library/Caches/fnm_multishells/76639_1710429675053/bin/yarn
  npm: 10.2.4 - ~/Library/Caches/fnm_multishells/76639_1710429675053/bin/npm
  pnpm: 8.15.4 - ~/Library/Caches/fnm_multishells/76639_1710429675053/bin/pnpm
NPM Packages:
  @aws-amplify/backend: 0.13.0-beta.5
  @aws-amplify/backend-cli: 0.12.0-beta.5
  aws-amplify: 6.0.20
  aws-cdk: 2.132.1
  aws-cdk-lib: 2.132.1
  typescript: 5.4.2
AWS environment variables:
  AWS_PROFILE = josef-gen2
  AWS_STS_REGIONAL_ENDPOINTS = regional
  AWS_NODEJS_CONNECTION_REUSE_ENABLED = 1
  AWS_SDK_LOAD_CONFIG = 1
No CDK environment variables
```

### Description

Cognito does not provide a raw URL, but rather allows customizing the message by replacing the inner contents of the shortcode
> When you choose Link as the verification type, your custom message must include a placeholder in the format `{##Verify Your Email##}`. You can change the text string between the placeholder characters, for example `{##Click here##}`. A verification link titled Verify Your Email replaces this placeholder.
https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-message-customizations.html#cognito-user-pool-settings-email-verification-message-customization

However the auth construct uses the default, static value `{##Verify Your Email##}` https://github.com/aws-amplify/amplify-backend/blob/main/packages/auth-construct/src/construct.ts#L63

This is especially important for i18n

**Comment by josefaidt** [comment]
https://www.typescriptlang.org/play?#code/MYewdgzgLgBApgWwIYEsA2AZFYDWAlOCENANzgCcYBeGACijgA8oB+ALhmnOwHMBKagD4YAAwDeAYgkASMQ2YwWLGACIAahRQAzAJ4wdIAK6VEqNCoC+UiyICwAKFCRYp9AGEQAEzgEipCtR0AlTC4lLWdvYOAPQAVLEOMLEwAMpwYJ4wSGDwyOgwAO4oUAAWWTBo2DiJsdEOUDoADnAwAKJ5aCkNaHBYuIEqGACSAHIA0iox8TWp6ZnZuWaFxWVIMKDeNXX2Dc1tHV06PR7eA24A8gAirZNRO00t7WYA6it9OIFiiZzdcBxP6EOPXeABpvgAjLw6Di0Sq4Di7OAgLSLdDvXzEMjkYLCLi8MH2CwOeoPfYvFYnFo0L72GA-I5-MmA36Ugl0yGeaF0DaM2g40SSCQXa4Rfl4sA8BxEu6ImAabhaFDAJBQFDgAFoAAK5BAjQggQ1r1K7xgAB8mWgjSVKSS9pc4IqwHAAIKGUra3X66nfNAgHjYK0cGl0umuNAI8iGFrm+XaJUqtVgDUevXfaXS4n2LSGMDAVXgGDeR0ut0lWiNHV6jj24uu92ViACYMwaLRGAAOk7UszTmgMHBSAgSrrZRoRewJdKtGbvv9YEDMGboY6EajbJg6b4PfAfaQpatGspgXHTpH059foDKyD32XZhvtJDdOgDI4KmFN3XT45XNoPP5Iixro+hGJQSDAKAOawEUpQwKUKD6jyHCyDyfI2LeG7rpu27OFk+4rBqACq5BoMeDoTmeM6XvO16LhhYYPk+z6-G+wzjCoX4hq2qhAXoBjGKi5gYeyUIwnCOD8uJfIYdKdLYXcvawHupQHh0xGWisAAqTCwGO5GnqW56PrOV6lIxIYMXRj5Pi+PSsaMEycXS3EqM8cBoKACAtFAIAwAAsjozqNI0ACEMCERA3klAhFRVHBvlYtofGgVkEFGGAUC3Ex-aiXQ4kAW5HkgF58X+YFwVhRFLSyFJKjwfq4kqHwFilYlwH8WBaVQZET6yRuUpbgpO5KfhpRESRQwAIwABw5HptaGVRc4LkugnmTZLGqGxjnCS2bZAfGcGGIJu0-mJVSSVUtDqJoirKkdglNTJaYDUAA

```ts
const emailLinkResolver = (text?: string) => `{##${text ?? "Verify your email"}##}`
const emailCodeResolver = () => `{####}`

/**
 * Send an email with a link
 */
type EmailStyleLink = "LINK"
/**
 * Send an email with a code
 */
type EmailStyleCode = "CODE"

type EmailWithLink = {
  style: EmailStyleLink,
  body: (link: typeof emailLinkResolver) => string,
}

type EmailWithCode = {
  style: EmailStyleCode,
  body: (code: () => `{##CODE##}`) => string
}

type VerificationEmailProps = EmailWithLink | EmailWithCode
type DefineAuthProps = {
  loginWith: {
    email: true | VerificationEmailProps
  }
}


function defineAuth(props: DefineAuthProps) {
  // ...
}

const basicAuth = defineAuth({
  loginWith: {
    email: true,
  }
})

const authWithEmailCode = defineAuth({
  loginWith: {
    email: {
      style: "CODE",
      body: (code) => `Verify your account with this code: ${code()}`
    },
  }
})

const authWithEmailUrl = defineAuth({
  loginWith: {
    email: {
      style: "LINK",
      // "Verify your email"
      body: (link) => link()
    }
  }
})

const authWithEmailUrlWithText = defineAuth({
  loginWith: {
    email: {
      style: "LINK",
      // "Welcome to MyApp! Use this link to verify your account"
      body: (link) => `Welcome to MyApp! Use ${link("this link")} to verify your account`
    }
  }
})

const authWithEmailUrlI18n = defineAuth({
  loginWith: {
    email: {
      style: "LINK",
      // Verifica tu email
      body: (link) => link("Verifica tu email")
    }
  }
})
```

**Comment by arundna** [comment]
This has been released on CLI `0.12.0-beta.17`
